src/main/java/org/broadinstitute/hellbender/utils/IndexUtils.java

```diff
@@ -0,0 +1,109 @@
+     * @return an Index, or null if we're unable to load
+     */
+    public static Index loadTabixIndex(final File featureFile) {
+        try {
+            final String path = featureFile.getAbsolutePath();
+            final boolean isTabix = new AbstractFeatureReader.ComponentMethods().isTabix(path, null);
+            if (! isTabix){
+                return null;
+            }
+            final String indexPath = ParsingUtils.appendToPath(path, TabixUtils.STANDARD_INDEX_EXTENSION);
+            logger.debug("Loading tabix index from disk for file " + featureFile);
+            final Index index = IndexFactory.loadIndex(indexPath);
+            final File indexFile = new File(indexPath);
+            checkIndexModificationTime(featureFile, indexFile, index);
+            return index;
+        } catch (final IOException e) {
+            return null;
+        }
+    }
+
+    /**
+     * @throws UserException the index is not the current version.
+     * Prints a warning if the index is not up-to date (ie older than the feature file)
+     */
+    public static void checkIndexModificationTime(final File featureFile, final File indexFile, final Index index) {
+        Utils.nonNull(featureFile, "featureFile");
+        Utils.nonNull(indexFile, "indexFile");
+        Utils.nonNull(index, "index");
+        if (! index.isCurrentVersion()) {
+            // we've loaded an old version of the index, we want to remove it <-- currently not used, but may re-enable
+            throw new UserException("Index file " + indexFile + " is out of date (old version). Use " + IndexFeatureFile.class.getSimpleName() +  " to make an index.");
+        } else if (indexFile.lastModified() < featureFile.lastModified()) {
+            logger.warn("Index file " + indexFile + " is out of date (index older than input file). Use " + IndexFeatureFile.class.getSimpleName() +  " to make a new index.");
+        }
+    }
+
+    /**
+     * get the sequence dictionary contig list that is in the index or null if there is no index or no contigs
+     * Note: the dictionary returned will not have the contig lengths filled in {@link SAMSequenceRecord#UNKNOWN_SEQUENCE_LENGTH} is used.
+     * Note: this method is specifically designed for getting sequence dictionaries from indices on Feature files (tribble or tabix indices)
+     * @return a SAMSequenceDictionary or null if the index cannot be loaded or there are no contigs in the index
+     */
+    public static SAMSequenceDictionary createSequenceDictionaryFromFeatureIndex(final File featureFile) {
+        final Index tribbleIndex = loadTribbleIndex(featureFile);
+        if (tribbleIndex != null) {
+            return getSamSequenceDictionaryFromIndex(tribbleIndex);
+        }
+
+        final Index tabixIndex = loadTabixIndex(featureFile);
+        if (tabixIndex != null){
+           return getSamSequenceDictionaryFromIndex(tabixIndex);
+        }
```

Comment:
Extract a public `loadIndex(featureFile)` method that calls both `loadTribbleIndex()` and `loadTabixIndex()` as appropriate, and call that new method from here. (It may be useful in the future to have a single/unified index-loading utility method)